File: spoj py/P153PROE.py
from sys import exit

n = long(input())

Pot = range(0,n+1)

s = raw_input()
s = s + " "
k = 0
i = 0
while i<len(s) and k<=n:
    temp = ""
    while s[i]==' ':
        i = i+1
    while s[i]!=' ':
        temp = temp + s[i]
        i = i+1
    Pot[k] = long(temp)
    k = k + 1

del i
del k
del s
del temp

i = n
carry = 0
while i>0:
    if Pot[i]%i==0:
        div = Pot[i]/i
        carry = carry + div
        Pot[i-1] = Pot[i-1] + carry
    else:
        print 'No'
        exit(0)
    i = i-1

print 'Yes'
exit(0)

File: P153PROE.cpp
#include<stdio.h>

int main(void)
{
	long n;
	scanf("%ld",&n);
	long long Pot[n];
	for(long i=0;i<=n;i++)
		scanf("%ld",&Pot[i]);
	long i=n;
	long long carry = 0, div;
	while(i>0)
	{
		if(Pot[i]%i==0)
		{
			div = Pot[i]/i;
			carry = carry + div; 
			Pot[i-1] += carry;
		}
		else
		{
			printf("No");
			return 0;
		}
		i--;
		
	}
	printf("Yes");
 return 0;
}

